From 41772bb7633fab5e38f0dac90e1e04538a066a3d Mon Sep 17 00:00:00 2001
From: Andrei Borzenkov <arvidjaar@gmail.com>
Date: Mon, 4 May 2015 09:13:53 +0300
Subject: [PATCH 161/260] Revert "efinet: memory leak on module removal"

This reverts commits 47b2bee3ef0ea60fc3f5bfc37f3784e559385297
and 8d3c4544ffdd0289a4b0bdeb0cdc6355f801a4b3. It is not safe
to free allocated cards, dangling pointers main remain. Such
cleanup requires more changes in net core.
---
 grub-core/net/drivers/efi/efinet.c | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/grub-core/net/drivers/efi/efinet.c b/grub-core/net/drivers/efi/efinet.c
index bb863e65a..78df215be 100644
--- a/grub-core/net/drivers/efi/efinet.c
+++ b/grub-core/net/drivers/efi/efinet.c
@@ -294,12 +294,6 @@ GRUB_MOD_FINI(efinet)
 
   FOR_NET_CARDS_SAFE (card, next) 
     if (card->driver == &efidriver)
-      {
-	grub_net_card_unregister (card);
-	grub_free (card->txbuf);
-	grub_free (card->rcvbuf);
-	grub_free ((char *)card->name);
-	grub_free (card);
-      }
+      grub_net_card_unregister (card);
 }
 
-- 
2.13.0

